pkg/provisioner/ironic/ironic.go

```diff
@@ -508,23 +508,30 @@ func (p *ironicProvisioner) InspectHardware() (result provisioner.Result, detail
 				result.RequeueAfter = introspectionRequeueDelay
 				err = nil
 				return
+			case nodes.Manageable:
+				inspect = true
 			default:
-				p.log.Info("starting new hardware inspection")
-				var success bool
-				success, result, err = p.tryChangeNodeProvisionState(
-					ironicNode,
-					nodes.ProvisionStateOpts{Target: nodes.TargetInspect},
-				)
-				if success {
-					p.publisher("InspectionStarted", "Hardware inspection started")
-				}
-				return
+				inspect = true
 			}
+		} else {
+			err = errors.Wrap(err, "failed to extract hardware inspection status")
+			return
+		}
+	}
+	if reinstrospect && !(nodes.ProvisionState(ironicNode.ProvisionState) == nodes.Inspecting || nodes.ProvisionState(ironicNode.ProvisionState) == nodes.InspectWait) {
+		p.log.Info("starting new hardware inspection")
+		var success bool
+		success, result, err = p.tryChangeNodeProvisionState(
+			ironicNode,
+			nodes.ProvisionStateOpts{Target: nodes.TargetInspect},
+		)
+		if success {
+			p.publisher("InspectionStarted", "Hardware inspection started")
 		}
-		err = errors.Wrap(err, "failed to extract hardware inspection status")
 		return
 	}
-	if !status.Finished {
+	if !status.Finished || (nodes.ProvisionState(ironicNode.ProvisionState) == nodes.Inspecting || nodes.ProvisionState(ironicNode.ProvisionState) == nodes.InspectWait) {
+		details = nil
 		p.log.Info("inspection in progress", "started_at", status.StartedAt)
```

Comment:
As an alternative to the inspect flag... we know the time at which the inspection started (and, from memory, ended if it is complete?), and we also know the time at which we entered the Inspecting state (from Host.Status.OperationHistory.Inspect.Start). So we could assume that if any existing inspection was done before we entered the inspecting state, then we need to start a new inspection.